infrastructure/tooling/src/smoketest/checks/secrets.js

```diff
@@ -0,0 +1,32 @@
+const taskcluster = require('taskcluster-client');
+const assert = require('assert');
+
+exports.tasks = [];
+exports.tasks.push({
+  title: 'Create secret/read-it-back check',
+  requires: [],
+  provides: [
+    'smoke-secret',
+  ],
+  run: async () => {
+    let secrets = new taskcluster.Secrets(taskcluster.fromEnvVars());
+    let baseUrl = 'project/taskcluster/smoketest/';
+    let secretName = taskcluster.slugid();
+    const payload = {
+      "expires": taskcluster.fromNowJSON('2 minutes'),
+      "secret": {
+        "description": secretName,
+        "type": "object",
+      },
+    };
+    await secrets.set(baseUrl, payload);
+    const getSecret = await secrets.get(baseUrl);
+    assert.deepEqual(getSecret.secret, payload.secret);
+    await secrets.remove(baseUrl);
+    await assert.rejects(
+      () => secrets.get(baseUrl),
+    ).then(()=>{
+      err => assert.equal(err.code, 404);
+    });
```

Comment:
This doesn't look quite right.  It's adding a callback to the Promise, and that callback just defines a function and the returns.  So it's not checking anything.  And, in fact, err.code is `ResourceNotFound` rather than 404.

```js
    await assert.rejects(
      () => secrets.get(baseUrl),
      err => assert.equal(err.code, 404)
    );
```

When writing tests, it's usually a good idea (and part of test-driven-development) to make sure the test can fail.  It's surprisingly easy to write a test that will always pass -- and such a test doesn't help anything!  One way I might check such a thing is to comment out the `secrets.remove` call and see that the last `secrets.get` doesn't return a 404.

Reply:
https://github.com/taskcluster/taskcluster/pull/1525#discussion_r333011807
Oh I see, the secret has to be appended to the baseUrl like the below code, right?
`project/taskcluster/smoketest/secretName` ,

Reply:
Yes, that's the correct secret name.  But I'd also like you to change the variable name "baseUrl" because it's not a URL.

Reply:
Yes, I will follow GitHub flow from next issue onward, as it is  best practise. I have also changed the variable name baseUrl and updated the pull request. I commented out `secrets.remove` and the last `secrets.get` does not return a 404, that means its working right?
@djmitche